lib/galaxy/model/scripts/user_deduplicator.py

```diff
@@ -0,0 +1,81 @@
+from sqlalchemy import (
+    func,
+    select,
+    update,
+)
+from sqlalchemy.engine import (
+    Connection,
+    Result,
+)
+
+from galaxy.managers.users import generate_next_available_username_with_connection
+from galaxy.model import User
+
+
+class UserDeduplicator:
+
+    def __init__(self, engine):
+        self.engine = engine
+
+    def deduplicate_emails(self) -> None:
+        with self.engine.begin() as connection:
```

Comment:
We can't do this, users won't control the new email address.

Reply:
As far as deduplication goes, there is not much else we can do: there should be no duplicate emails in the database, and if there are, we can either delete the duplicate records or edit the duplicate values. Another option is to expect admins to do this manually (deleting rows, contacting users, whatever), and add this to the admin release notes. This script modifies the emails on the older duplicate accounts, leaving the newest intact, which of course is not ideal.

Reply:
> there is not much else we can do:

At a minimum set the email address to something not guessable ? Yes, if there are duplicates it's better to check which account actually has data, it's not unlikely only one of them will have anything at all, since we probably pick one on login, or we've always been denying / failing at the login stage, at which point toggling the delete flag is sufficient.